### PR TITLE
Add doc comments to "unsafe" `AsSpan()` and indexer members

### PR DESCRIPTION
**Pull request by AArnott**

### CODE REVIEW

File: src/Microsoft.Windows.CsWin32/Generator.cs [hunk 1]
```diff
@@ -82,6 +82,20 @@ public class Generator : IDisposable
 
         private const string SimpleFileNameAnnotation = "SimpleFileName";
         private const string OriginalDelegateAnnotation = "OriginalDelegate";
+        private static readonly SyntaxTriviaList InlineArrayUnsafeAsSpanComment = ParseLeadingTrivia(@"/// <summary>
+/// Gets this inline array as a span.
+/// </summary>
+/// <remarks>
+/// Important: When this struct is on the stack, do not let the returned span outlive the stack frame that defines it.
```
Comment (jnm2) [inline src/Microsoft.Windows.CsWin32/Generator.cs:89]: I sometimes use the '⚠' character in XML docs.

Comment (AArnott) [reply src/Microsoft.Windows.CsWin32/Generator.cs:89]: Sounds good.

**Comment by AArnott** [comment]
oh nuts. The multi-line text comparison is rearing its ugly head. Test fails on linux due to inconsistent line endings.

**Comment by jnm2** [comment]
...a good thing the tests are there, right? Otherwise you might be generating source with inconsistent line endings. 😁

**Pull request merged**